testing/internal/e2e/boundary/role.go

```diff
@@ -70,4 +80,59 @@ func AddPrincipalToRoleCli(t testing.TB, ctx context.Context, roleId string, pri
 		),
 	)
 	require.NoError(t, output.Err, string(output.Stderr))
+	t.Logf("Principle %s added to role: %s", principal, roleId)
+}
+
+// SetGrantScopesToRoleCli uses Boundary CLI to override grant scopes for the role with the provided ones.
+// Option WithGrantScopeId can be used multiple times to provide grant scope IDs.
+func SetGrantScopesToRoleCli(t testing.TB, ctx context.Context, roleId string, opt ...RoleOption) error {
+	opts := getRoleOpts(opt...)
+	var args []string
+
+	if len(opts.scopeIds) == 0 {
+		return fmt.Errorf("at least one grant scope id must be provided")
```

Comment:
Not sure what my opinion is yet, but a thought I was having was that it felt weird to have this required parameter that is optional in the method definition

Reply:
I had the same feeling and checked what approach we had in the code base. I found that we used options for required fields when using api clients, e.g. names for targets

```go
newTargetResult, err := tClient.Create(ctx, "tcp", projectId,
		targets.WithName("e2e Target"),
		targets.WithTcpTargetDefaultPort(uint32(targetPort)),
)
```

Alternatively, I thought about adding an array of grant scopes instead of options. I don't dislike it either 🙂

Reply:
I see. If it's an existing pattern, then we can keep as-is.